app/publish/tus_test.go

```diff
@@ -0,0 +1,429 @@
+package publish
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/lbryio/lbrytv/app/auth"
+	"github.com/lbryio/lbrytv/app/sdkrouter"
+	"github.com/lbryio/lbrytv/app/wallet"
+	"github.com/lbryio/lbrytv/apps/lbrytv/config"
+	"github.com/lbryio/lbrytv/internal/errors"
+	"github.com/lbryio/lbrytv/internal/test"
+	"github.com/lbryio/lbrytv/models"
+)
+
+func newTestPublishHandler(t *testing.T) *PublishHandler {
+	t.Helper()
+
+	h, err := NewHandler(
+		newTestAuthProvider(),
+		os.TempDir(),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return h
+}
+
+func newTestAuthProvider() auth.Provider {
+	return auth.NewIAPIProvider(
+		sdkrouter.New(config.GetLbrynetServers()),
+		config.GetInternalAPIHost(),
+	)
+}
+
+// tr is a helper to construct publish route path
+var tr testRoute = "/api/v2/publish"
+
+type testRoute string
+
+func (tr testRoute) withID(id int) string {
+	return fmt.Sprintf("%s/%d", tr, id)
+}
+
+func (tr testRoute) notify(id int) string {
+	return fmt.Sprintf("%s/%d/notify", tr, id)
+}
+
+func (tr testRoute) root() string {
+	return fmt.Sprintf("%s/", tr)
+}
+
+func newTestProvider(token, ip string) (*models.User, error) {
+	reqChan := test.ReqChan()
+	ts := test.MockHTTPServer(reqChan)
+	var u *models.User
+	if token == "uPldrToken" {
+		u = &models.User{ID: 20404}
+		u.R = u.R.NewStruct()
+		u.R.LbrynetServer = &models.LbrynetServer{Address: ts.URL}
+	}
+	go func() {
+		ts.NextResponse <- expectedStreamCreateResponse
+	}()
+	return u, nil
+}
+
+func newTestMux(h *PublishHandler, mwf ...mux.MiddlewareFunc) *mux.Router {
+	router := mux.NewRouter()
+
+	testRouter := router.PathPrefix("/api/v2/publish").Subrouter()
+	for _, fn := range mwf {
+		testRouter.Use(fn)
+	}
+
+	testRouter.HandleFunc("/", h.PostFile).Methods(http.MethodPost)
+	testRouter.HandleFunc("/{id}", h.HeadFile).Methods(http.MethodHead)
+	testRouter.HandleFunc("/{id}", h.PatchFile).Methods(http.MethodPatch)
+	testRouter.HandleFunc("/{id}/notify", h.Notify).Methods(http.MethodPost)
+
+	return testRouter
+}
+
+func newPartialUpload(t *testing.T, h *PublishHandler) string {
+	t.Helper()
+
+	testData := []byte("test file")
+	r, err := http.NewRequest(http.MethodPost, tr.root(), bytes.NewReader(testData))
+	if err != nil {
+		t.Fatalf("failed to create a new request: %v", err)
+	}
+	r.Header.Set(wallet.TokenHeader, "uPldrToken")
+	r.Header.Set("Upload-Length", fmt.Sprintf("%d", len(testData)))
+
+	w := httptest.NewRecorder()
+	newTestMux(h, auth.Middleware(newTestProvider)).ServeHTTP(w, r)
+
+	resp := w.Result()
+	if resp.StatusCode != http.StatusCreated {
```

Comment:
`github.com/stretchr/testify/assert` or `require` works really well to get rid of those conditionals, we use it everywere in the test code.

Reply:
gotcha, i prefer to use standard library as much as possible, and i don't mind more verbosity on the tests actually
but yea happy to change so it follows the rest of the test code

Reply:
Agree with using the standard library but tests are normally already verbose, every bit that helps to make them easier to read is helpful.